stubs/docutils/docutils/parsers/rst/__init__.pyi

```diff
@@ -16,6 +17,18 @@ class DirectiveError(Exception):
     def __init__(self, level: Any, message: str) -> None: ...
 
 class Directive:
+    def __init__(
+        self,
+        name: str,
+        arguments: List[object],
+        options: Dict[str, object],
+        content: List[str],
```

Comment:
Normally, as above, we prefer ABCs over concrete classes for argument types, but as the arguments are directly assigned to fields, I can see the point in enforcing concrete types here. But we should use `Any` instead of `object` here. The latter indicates that literally any object is accepted, i.e. `arguments` could be `["Foo", 3]`, while the former is a marker that the type can't be represented accurately with the current type system.

Also, we prefer built-in generics now in typeshed:

```suggestion
        arguments: list[Any],
        options: dict[str, Any],
        content: list[str],
```

Reply:
This was a distinction that I was entirely unaware of! I'll keep these as they are and change the ones you've suggested to abstract base classes.

Reply:
Done and swapped to Any.

Reply:
For future reference, you also need to consider invariance if you want to use `list[object]`. For example, this does **not** type-check:

```python
def foo(obj: list[object]) -> None:
    pass

obj: list[int] = [1]
foo(obj)
```

An instance of `list[int]` is not a valid `list[object]` (in other words, `list` is invariant). If a `list[int]` was a valid `list[object]`, then `foo()` could do `obj.append("lol")`, which is valid since `list[object]` can contain any objects, but then you would end up with an instance of `list[int]` that contains `"lol"`.

The same applies to other mutable containers too. For example, a `dict[str, str]` is not a valid `dict[str, object]`.

To work around this, you can use `list[Any]`, which means that a list with any item type is accepted. Here `Any` means "no type errors please", which is not same as "accept any object".

Reply:
That's really interesting, I had no idea about that! Thank you for pointing that out.
I was having trouble replicating this as I was going straight for 
```
foo([1])
```
which mypy doesn't complain about interestingly which is possibly why I've got away with when using covariant arguments in the past.

Any idea why 
```
foo(a=[1])
```
passes `$ mypy types.py`
and 
```
o: list[int] = [1]
foo(o)
```
gives me
```
$ mypy types.py
types.py:15: error: Argument 1 to "foo" has incompatible type "List[int]"; expected "List[object]"
types.py:15: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
types.py:15: note: Consider using "Sequence" instead, which is covariant
```
?
(This is _well_ outside the scope of the PR btw so feel free to tell me to RTFM 😂)

Reply:
Mypy special-cases the `[` `]` syntax. It can interpret `[1]` as `list[object]` or `list[int]`, whichever works better. With a type-annotated variable, you force it to decide it is `list[int]`.

Reply:
Aaah makes sense, thanks again for the explanation 👍